src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         }
     }
 
+    private boolean isProxyRequested() {
+        // HTTPS requests have uri without http scheme too
+        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
+            return false;
+        }
+        // direct requests to the proxy have the path only without a scheme
```

Comment:
Every local request to the proxy has no scheme, host and port in the 
URI. It's provided by LittleProxy or Netty somewhere, but HTTPS requests 
have a similiar behavior.

Please have a look at ganskef/LittleProxy-mitm and ganskef/LittleProxy. 
I've added some documentation in the README's regarding this.

Am 20.07.2015 um 22:02 schrieb Jason Hoetger:
> In src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java
> <https://github.com/adamfisk/LittleProxy/pull/220#discussion_r35037070>:
>
>> @@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
>>          }
>>      }
>>
>> +    private boolean isProxyRequested() {
>> +        // HTTPS requests have uri without http scheme too
>> +        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
>> +            return false;
>> +        }
>> +        // direct requests to the proxy have the path only without a scheme
>
> Where is it actually comparing the URI to localhost/127.0.0.1/etc.?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adamfisk/LittleProxy/pull/220/files#r35037070>.
>